api/v1alpha1/gateway_types.go

```diff
@@ -68,18 +73,43 @@ const (
 	HTTPSProcotol = "HTTPS"
 )
 
+// Route defines the schema for a route.
+type Route struct {
+	// RouteRef is a reference to an object to associate with the Gateway.
+	// RouteRef defines protocol-specific routing to back-ends (e.g. Services).
```

Comment:
Is this really what a Route is? Looking at the current spec, `TcpRoute` and `HTTPRoute` are defined. To my mind these have more in common with endpoints (since we are in a reverse-proxy configuration, they will appear to be endpoints from the perspective of the client).

I realize that this terminology is already in play, but maybe we can clarify here. The description of "routing to back-ends" seems more applicable to `HTTPRouteAction`.

Reply:
Can you define more what "endpoint" and "client" you mention in the above.
Endpoint e.g. k8s Endpoint? Or something else?
Client as in the HTTP-client coming to the Gateway (or something else)

Reply:
The purpose of the `Route` type is to group attributes with the route reference. Currently, we're focused on TLS, but I can foresee other attributes a Cluster Operator may want to associate to a route, e.g. a traffic shaping policy.